### PR TITLE
Obsolete the host's Roles 

### PR DESCRIPTION
**Pull request by seanfarmar**
#127

### CODE REVIEW

File: src/NServiceBus.Hosting.Tests/APIApprovals.Approve.approved.txt [hunk 1]
```diff
@@ -5,7 +5,9 @@
 [assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5.2", FrameworkDisplayName=".NET Framework 4.5.2")]
 namespace NServiceBus
 {
+    [System.ObsoleteAttribute(@"Roles are obsoleted and should not be used. The functionality of AsA_Server, and AsA_Publisher has been made defaults in the core, see: https://docs.particular.net/nservicebus/hosting/nservicebus-host/?version=host_6#roles-built-in-configurations. Will be treated as an error from version 7.1.0. Will be removed in version 8.0.0.", false)]
     public interface AsA_Client { }
+    [System.ObsoleteAttribute(@"Roles are obsoleted and should not be used. The functionality of AsA_Server, and AsA_Publisher has been made defaults in the core, see: https://docs.particular.net/nservicebus/hosting/nservicebus-host/?version=host_6#roles-built-in-configurations. Will be treated as an error from version 7.1.0. Will be removed in version 8.0.0.", false)]
```
Comment (SimonCropp) [inline src/NServiceBus.Hosting.Tests/APIApprovals.Approve.approved.txt]: no guarantee this doco url will exist for the same period that this error might be seen by users

File: src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs [hunk 2]
```diff
@@ -4,5 +4,10 @@ namespace NServiceBus
     /// Indicates this endpoint is a client.
     /// As such will be set up as a non-transactional endpoint with no impersonation and purging messages on startup.
     /// </summary>
+
+    [ObsoleteEx(
+    TreatAsErrorFromVersion = "7.1",
+    Message = "Roles are obsoleted and should not be used. The functionality of AsA_Server, and AsA_Publisher has been made defaults in the core, see: https://docs.particular.net/nservicebus/hosting/nservicebus-host/?version=host_6#roles-built-in-configurations",
```
Comment (SimonCropp) [inline src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs]: > Roles are obsoleted and should not be used. 

this is redundant

**Comment by seanfarmar** [comment]
Made changes based on @SimonCropp comments

**Comment by seanfarmar** [comment]
@SimonCropp @andreasohlund @Particular/host-maintainers happy to merge?

File: src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs [hunk 3]
```diff
@@ -4,5 +4,10 @@ namespace NServiceBus
     /// Indicates this endpoint is a client.
     /// As such will be set up as a non-transactional endpoint with no impersonation and purging messages on startup.
     /// </summary>
+
+    [ObsoleteEx(
+    TreatAsErrorFromVersion = "7.1",
+    Message = "The functionality of AsA_Server, and AsA_Publisher has been made defaults in the core.",
```
Comment (ramonsmits) [inline src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs]: @seanfarmar This message seems incorrect.

File: src/NServiceBus.Hosting.Windows/Roles/AsA_Server.cs [hunk 4]
```diff
@@ -4,5 +4,10 @@ namespace NServiceBus
     /// Indicates this endpoint is a server.
     /// As such will be set up as a transactional endpoint using impersonation, not purging messages on startup.
     /// </summary>
+
+    [ObsoleteEx(
+        TreatAsErrorFromVersion = "7.1",
+        Message = "The functionality of AsA_Server, and AsA_Publisher has been made defaults in the core.",
```
Comment (ramonsmits) [inline src/NServiceBus.Hosting.Windows/Roles/AsA_Server.cs]: @seanfarmar This message seems incorrect.

**Comment by ramonsmits** [comment]
@seanfarmar Also, why is this marked as obsolete as develop would already be 8.0.0 if I'm not mistaken. I think they should be marked as:

```c#
    [ObsoleteEx(
        TreatAsErrorFromVersion = "8.0",
        Message = "Configure logging in the constructor of the class that implements IConfigureThisEndpoint.",
        RemoveInVersion = "8.0")]
```

**Comment by ramonsmits** [comment]
@seanfarmar Would this become part of 8.0.0? Should we add it to a milestone?

File: src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs (hunk 3 above)
Comment (seanfarmar) [inline src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs]: What is wrong with the message?

File: src/NServiceBus.Hosting.Windows/Roles/AsA_Server.cs (hunk 4 above)
Comment (seanfarmar) [inline src/NServiceBus.Hosting.Windows/Roles/AsA_Server.cs]: What is wrong with the message?

Comment (ramonsmits) [inline src/NServiceBus.Hosting.Windows/Roles/AsA_Server.cs]: @seanfarmar 
> has been made defaults

File: src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs [hunk 5]
```diff
@@ -4,5 +4,10 @@ namespace NServiceBus
     /// Indicates this endpoint is a client.
     /// As such will be set up as a non-transactional endpoint with no impersonation and purging messages on startup.
     /// </summary>
+
+    [ObsoleteEx(
+    TreatAsErrorFromVersion = "7.1",
```
Comment (ramonsmits) [inline src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs]: Must be I think 8.0

File: src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs (hunk 3 above)
Comment (ramonsmits) [inline src/NServiceBus.Hosting.Windows/Roles/AsA_Client.cs]: "The AsA_Server, and AsA_Publisher roles are obsoleted. Manually configure the EndpointConfiguration object via IConfigureThisEndpoint.Customize(EndpointConfiguration endpointConfiguration)"

File: src/NServiceBus.Hosting.Windows/Roles/AsA_Server.cs [hunk 6]
```diff
@@ -4,5 +4,10 @@ namespace NServiceBus
     /// Indicates this endpoint is a server.
     /// As such will be set up as a transactional endpoint using impersonation, not purging messages on startup.
     /// </summary>
+
+    [ObsoleteEx(
+        TreatAsErrorFromVersion = "7.1",
```
Comment (ramonsmits) [inline src/NServiceBus.Hosting.Windows/Roles/AsA_Server.cs]: 8.0

**Comment by bording** [comment]
@ramonsmits Looks like another place where if you're changing the `TreatAsErrorFromVersion` to 8, you need to bump the `RemoveInVersion` to 9.

**Comment by seanfarmar** [comment]
@ramonsmits please review

**Comment by andreasohlund** [comment]
@Particular/host-maintainers we're going dotnet core beta 1 this week, should this be included in the core beta? (needs a rebase and approval from @ramonsmits )

**Comment by andreasohlund** [comment]
I've added this to the upgrade guide in this PR

https://github.com/Particular/docs.particular.net/pull/3173

**Comment by seanfarmar** [comment]
@ramonsmits can you review this and let me know if there is more to be done here?

File: src/NServiceBus.Hosting.Windows/GenericHost.cs [hunk 7]
```diff
@@ -87,12 +87,33 @@ Task<IStartableEndpoint> PerformConfiguration(Action<EndpointConfiguration> more
             moreConfiguration?.Invoke(configuration);
 
             specifier.Customize(configuration);
-            RoleManager.TweakConfigurationBuilder(specifier, configuration);
+
+            Type transportDefinitionType;
+            if (TryGetTransportDefinitionType(specifier, out transportDefinitionType))
```
Comment (andreasohlund) [inline src/NServiceBus.Hosting.Windows/GenericHost.cs]: `UsingTransport` is also a role, lets get rid of it as well?

**Comment by andreasohlund** [comment]
Replaced by #136

**Pull request closed without merging**